File: javascript/mathNet/Distributions/Normal.js




class Normal{}

Normal.CDF=function(mean, stddev, x){
    return 0.5 * SpecialFunctions.Erfc((mean - x) / (stddev * Constants.Sqrt2));
}
Normal.InvCDF=function(mean, stddev, p){
    return mean - (stddev * Constants.Sqrt2 * SpecialFunctions.ErfcInv(2.0 * p));
}

Normal.PDF=function(mean, stddev, x){
    var d = (x - mean) / stddev;
    return Math.exp(-0.5 * d * d) / (Constants.Sqrt2Pi * stddev);
}


module.exports=Normal;

File: csharp/ToolGood.Algorithm2/MathNet/Distributions/Normal.cs
using System;

namespace ToolGood.Algorithm.MathNet.Numerics
{
    internal sealed class Normal
    {
        public static double CDF(double mean, double stddev, double x)
        {
            //if (stddev < 0.0) {
            //    throw new ArgumentException(Resources.InvalidDistributionParameters);
            //}

            return 0.5 * SpecialFunctions.Erfc((mean - x) / (stddev * Constants.Sqrt2));
        }

        public static double InvCDF(double mean, double stddev, double p)
        {
            //if (stddev < 0.0) {
            //    throw new ArgumentException(Resources.InvalidDistributionParameters);
            //}

            return mean - (stddev * Constants.Sqrt2 * SpecialFunctions.ErfcInv(2.0 * p));
        }

        public static double PDF(double mean, double stddev, double x)
        {
            //if (stddev < 0.0) {
            //    throw new ArgumentException(Resources.InvalidDistributionParameters);
            //}

            var d = (x - mean) / stddev;
            return Math.Exp(-0.5 * d * d) / (Constants.Sqrt2Pi * stddev);
        }
    }
}